datafusion/expr/src/expr_rewriter.rs

```diff
@@ -525,29 +525,55 @@ pub fn coerce_plan_expr_for_schema(
     plan: &LogicalPlan,
```

Comment:
I was somewhat confused about this at first because the name of this function 

it turns out it is only called for coercing union schemas: https://github.com/search?q=repo%3Aapache%2Farrow-datafusion%20coerce_plan_expr_for_schema&type=code

👍